### DISCUSSION TITLE
Question about fix_formula

### DISCUSSION THREAD

**Discussion by afedynak** [FrederickHuangLin/ANCOMBC#135] [Q&A]
Can someone clarify the parameter 'fix_formula' and how it should be different than 'group'

**Comment by FrederickHuangLin** [comment]
Hi @afedynak,

`fix_formula` is the character string expresses how the microbial absolute abundances for each taxon depend on the fixed effects in metadata. For instance, if you are interested in the impacts of `age` and `bmi`, you can specify `fix_formula = age + bmi`. If your variable of interest is continuous, such as `age`, that's it, you do not have to specify the `group` argument.

However, if your variable of interest is categorical, for instance, `bmi`, which contains `lean`, `overweight`, and `obese` categories. You may want to specify both `fix_formula = age + bmi` and `group = bmi` to let `ancombc` or `ancombc2` function know that you are interested in performing group comparisons (pairwise test, trend test, etc). `group` argument is optional unless the multi-group comparison or presence/absence test is desired.

Best,
Huang

**Comment by ElizabethOLeary** [comment]
Hello Huang, 

I have a similar question to the above, except I have two categorical variables of interest, "Treatment," which can be either "Placebo" or Probiotic," and "Diet," which can be either "Standard" or "Western." 

Presently, I am unable to find a way to run the ancombc2 function such that it will perform the multi-group comparison. Is there a way to do this with two categorical variables?

Thank you for any insight you can provide! 

Just FYI, here is my code (it will only run with one categorical variable in "group". If I run group = 'Treatment+Diet' I get the following error message: Error: subscript contains invalid names.

out_ta6_txd = ancombc2(data = physeq, assay_name = "counts", fix_formula = 'Treatment+Diet',
              tax_level = "ta6", 
              p_adj_method = "holm", prv_cut = 0.10, lib_cut = 1000, 
              group = 'Diet', struc_zero = TRUE, neg_lb = TRUE, 
              alpha = 0.05, global = TRUE, 
              n_cl = 1, verbose = TRUE)
